resources/assets/components/SignupCard/index.js

```diff
@@ -1,92 +1,66 @@
 import React from 'react';
-import { map } from 'lodash';
-import PropTypes from 'prop-types';
-import './signup-card.scss';
 import PostTile from '../PostTile';
-import { RestApiClient } from '@dosomething/gateway';
-import { extractPostsFromSignups } from '../../helpers';
 
-class SignupCard extends React.Component {
-  render() {
-    const signup = this.props.signup;
-    const campaign = this.props.campaign;
-    const gallerySize = 4;
+import './signup-card.scss';
 
-    const extraPostCount = signup.posts.data.length - gallerySize;
+const POST_GALLERY_SIZE = 4;
 
-    const posts = signup.posts.data
-      .slice(0, gallerySize)
-      .map(post => <PostTile key={post.id} post={post} />);
+const SignupCard = ({ signup }) => {
+  const campaign = signup.campaign || {};
 
-    const signupUrl = `/signups/${signup.id}`;
+  const extraPostCount = signup.posts.length - POST_GALLERY_SIZE;
+  const posts = signup.posts.slice(0, POST_GALLERY_SIZE).map(post => (
+    <li key={post.id}>
+      <PostTile post={post} />
+    </li>
+  ));
 
-    return (
-      <article className="container__row signup-card">
-        <a href={signupUrl}>
-          <div className="container__block -half">
-            <div className="container__row">
-              <h2 className="heading">
-                {campaign ? campaign.internal_title : signup.campaign_id}
-              </h2>
-              <h4 className="heading">Campaign ID: {signup.campaign_id}</h4>
-              {campaign ? (
-                <h4 className="heading">
-                  Campaign Start Date: {campaign.start_date}
-                </h4>
-              ) : null}
-            </div>
-            <div className="container__row">
-              <h4 className="heading">Why Statement</h4>
-              <p>{signup.why_participated}</p>
-            </div>
+  return (
+    <article className="container__row signup-card">
+      <a href={`/signups/${signup.id}`}>
+        <div className="container__block -half">
+          <div className="container__row">
+            <h2 className="heading">
+              {campaign.internalTitle || 'Unknown Campaign'}
```

Comment:
I would have thought this would just return a boolean, is this a shortcut to get to something like `campaign.internalTitle ? campaign.internalTitle : 'Unknown Campaign`?

Reply:
Yep! It's a trick that takes advantage of the fact that `||` will short-circuit once it gets something "truthy", and so if `campaign.internalTitle` exists it'll do that, and only evaluate the second part of the expression if not.

Reply:
Once it gets something truthy won't it return `true` though? Sorry this is blowing my mind a bit 🐡

Reply:
Nope! It just returns the "truthy" value. This used to be [a common, if somewhat flawed, way to set default values](https://www.codereadability.com/javascript-default-parameters-with-or-operator/) before that became part of the language (with default parameters for functions).

Reply:
> First of all, it isn’t obvious why the OR operator would be used to assign default values. It takes time for this hack to become intuitive.

This sums up my feelings 😂 but it does work! Very cool!